From d23e9b88671d72caffa970f6176fab09d381bfd9 Mon Sep 17 00:00:00 2001
From: Michal Srb <msrb@suse.com>
Date: Thu, 5 Apr 2018 14:48:46 +0200
Subject: [PATCH] cmake: Trust pkg-config in FindJsoncpp.

If it says that the module was found, then accept it even if
JSONCPP_INCLUDE_DIRS or JSONCPP_CFLAGS are empty. They may be empty if jsoncpp
is installed in system directories.
---
 cmake/FindJsoncpp.cmake | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/cmake/FindJsoncpp.cmake b/cmake/FindJsoncpp.cmake
index cee44d10..663288ca 100644
--- a/cmake/FindJsoncpp.cmake
+++ b/cmake/FindJsoncpp.cmake
@@ -1,6 +1,7 @@
 find_package(PkgConfig QUIET)
 if (PkgConfig_FOUND)
   pkg_search_module(JSONCPP jsoncpp QUIET)
+  set(Jsoncpp_FOUND JSONCPP_FOUND)
 
   # Get the libraries full paths, to be consistent with find_library().
   set(fulllibs)
@@ -38,10 +39,10 @@ else ()
   find_library(JSONCPP_LIBRARY NAMES jsoncpp)
   mark_as_advanced(JSONCPP_LIBRARY)
   set(JSON_LIBRARIES ${JSON_LIBRARY})
-endif ()
 
-include(FindPackageHandleStandardArgs)
-find_package_handle_standard_args(Jsoncpp DEFAULT_MSG JSONCPP_INCLUDE_DIRS JSONCPP_LIBRARIES JSONCPP_DEFINITIONS)
+  include(FindPackageHandleStandardArgs)
+  find_package_handle_standard_args(Jsoncpp DEFAULT_MSG JSONCPP_INCLUDE_DIRS JSONCPP_LIBRARIES JSONCPP_DEFINITIONS)
+endif ()
 
 if (Jsoncpp_FOUND)
   add_library(jsoncpp INTERFACE IMPORTED)
-- 
2.13.6

